src/hotspot/share/opto/callnode.hpp

```diff
@@ -542,6 +607,20 @@ class SafePointScalarObjectNode: public TypeNode {
 
   virtual uint size_of() const { return sizeof(*this); }
 
+  bool is_from_merge() const { return _merge_pointer_idx >= 0; }
+
+  int merge_pointer_idx(JVMState* jvms) const {
+    assert(jvms != nullptr, "JVMS reference is null.");
+    return jvms->scloff() + _merge_pointer_idx;
```

Comment:
how about we also assert is_from_merge() here? 

Your comment above says that _merge_point_idx is a zero-based index of sfpt's input array. 
here we use scloff-based. I think either is okay, but we need consistency.

Reply:
I think adding assert is a good idea. I'll fix the comment. Thanks!